atc/atccmd/command.go

```diff
@@ -445,22 +447,21 @@ func (cmd *RunCommand) Runner(positionalArguments []string) (ifrit.Runner, error
 	}
 
 	// Prepare distributed tracing facilities
+	var exp export.SpanSyncer
 	switch {
 	case cmd.Tracing.Jaeger.IsConfigured():
-		exp, err := cmd.Tracing.Jaeger.Exporter()
-		if err != nil {
-			return nil, err
-		}
-
-		tracing.ConfigureTracer(exp)
+		exp, err = cmd.Tracing.Jaeger.Exporter()
 	case cmd.Tracing.Stackdriver.IsConfigured():
-		exp, err := cmd.Tracing.Stackdriver.Exporter()
-		if err != nil {
-			return nil, err
-		}
-
-		tracing.ConfigureTracer(exp)
+		exp, err = cmd.Tracing.Stackdriver.Exporter()
+	}
+	if err != nil {
+		return nil, err
+	}
+	tp, err := tracing.TraceProvider(exp)
+	if err != nil {
+		return nil, fmt.Errorf("failed to configure trace provider: %w", err)
 	}
+	tracing.ConfigureTraceProvider(tp)
```

Comment:
I'm not quite getting this change - assuming `tracing.TraceProvider` is fine with having a `nil` `export.SpanSyncer`, are we not still always setting `tracing.Configured` to `true` (even when neither Jaeger nor Stackdriver are configured)?